ts/protocol/utils.ts

```diff
@@ -192,3 +192,21 @@ export function timeoutPromise<T>(error: Error, timeoutMs: number): Promise<T> {
   let promiseResolver = new TimeoutPromiseResolver(error, timeoutMs);
   return promiseResolver.promise;
 }
+
+/**
+ * Is a cancellable PromiseResolver.
+ */
+export class CancellablePromise<T> extends PromiseResolver<T> {
+  private static CANCELLED = 'CancellablePromise { Operation Cancelled }';
+
+  // the error returned when a cancellable promise is cancelled.
+  static CANCELLED_ERROR: Error&never = new Error(CancellablePromise.CANCELLED);
```

Comment:
I went through the discussion you had yesterday, but I thought the `never` was meant to be used as the type of CancellablePromise's template. Why using &never here?